src/main/java/uk/gov/companieshouse/company/profile/service/CompanyProfileService.java

```diff
@@ -54,21 +61,19 @@ public CompanyProfileService(Logger logger, CompanyProfileRepository companyProf
      */
     public void updateInsolvencyLink(final CompanyProfile companyProfileRequest)
             throws NoSuchElementException {
-
         Optional<CompanyProfileDocument> companyProfileOptional = companyProfileRepository
-                .findCompanyProfileDaoByCompanyProfile_Data_CompanyNumber(
-                        companyProfileRequest.getData().getCompanyNumber());
+                .findById(companyProfileRequest.getData().getCompanyNumber());
 
         if (companyProfileOptional.isEmpty()) {
             throw new NoSuchElementException("Database entry not found");
         }
 
         CompanyProfileDocument companyProfile = companyProfileOptional.get();
         String insolvencyLink = companyProfileRequest.getData().getLinks().getInsolvency();
-        companyProfile.companyProfile.getData().getLinks().setInsolvency(insolvencyLink);
+        companyProfile.companyProfile.getLinks().setInsolvency(insolvencyLink);
         companyProfileRepository.save(companyProfile);
```

Comment:
Not sure if this is persisting properly without WritingConverter, or is it handled in a different story?